core/src/test/java/org/infinispan/scattered/statetransfer/CoordStopTest.java

```diff
@@ -0,0 +1,166 @@
+            new CoordStopTest().biasAcquisition(BiasAcquisition.ON_WRITE)
+      };
+   }
+
+   @Override
+   protected void createCacheManagers() throws Throwable {
+      ConfigurationBuilder cb = new ConfigurationBuilder();
+      cb.clustering().cacheMode(CacheMode.SCATTERED_SYNC).hash().numSegments(3);
+      if (biasAcquisition != null) {
+         cb.clustering().biasAcquisition(biasAcquisition);
+      }
+      ControlledConsistentHashFactory.Scattered chf = new ControlledConsistentHashFactory.Scattered(new int[] {0, 1, 2});
+      cb.clustering().hash().consistentHashFactory(chf);
+
+      addClusterEnabledCacheManager(cb);
+      addClusterEnabledCacheManager(cb);
+      // If the updated topologies from old coord come when it's no longer in the view these are ignored.
+      // Therefore we have to delay the view.
+      GlobalConfigurationBuilder gcb = new GlobalConfigurationBuilder();
+      gcb.transport().transport(new DelayedViewJGroupsTransport(viewLatch));
+      addClusterEnabledCacheManager(gcb, cb);
+
+      assertTrue(cache(0).getCacheManager().isCoordinator());
+      // start other caches
+      cache(1);
+      cache(2);
+      waitForClusterToForm();
+
+      // segment 0 will be moved to cache(2). Since we've lost coord cache(1) now -> 0 and cache(2) -> 1
+      chf.setOwnerIndexes(new int[][] { {1}, {0}, {1} });
+   }
+
+   @AfterMethod
+   @Override
+   protected void clearContent() throws Throwable {
+      super.clearContent();
+      viewLatch = new CountDownLatch(1);
+   }
+
+   // Reproducer for ISPN-9128
+   public void testCoordLeaves() throws InterruptedException, ExecutionException {
+      MagicKey key = new MagicKey(cache(0));
+      cache(1).put(key, "value");
+      int stableTopologyId = cache(1).getAdvancedCache().getDistributionManager().getCacheTopology().getTopologyId();
+      BlockingLocalTopologyManager ltm2 = BlockingLocalTopologyManager.replaceTopologyManager(manager(2), cache(2).getName());
+      ControlledRpcManager rpcManager2 = ControlledRpcManager.replaceRpcManager(cache(2));
+      rpcManager2.excludeCommands(cmd -> cmd instanceof StateResponseCommand, CoordStopTest::isCancelStateTransfer);
+
+      log.infof("Stopping coordinator %s, last stable topology is %d", manager(0), stableTopologyId);
+      manager(0).stop();
+
+      log.infof("Current members are: %s", cache(1).getCacheManager().getMembers());
+      assertTrue(cache(1).getCacheManager().isCoordinator());
+
+      // topology + 1 is the one that just omits the leaving node
+      BlockedTopology t1 = ltm2.expectTopologyUpdate(CacheTopology.Phase.NO_REBALANCE, stableTopologyId + 1);
+      assertEquals(2, t1.getCacheTopology().getActualMembers().size());
+      assertEquals(null, t1.getCacheTopology().getPendingCH());
+      for (Address member : t1.getCacheTopology().getMembers()) {
+         assertEquals("Topology: " + t1.getCacheTopology(), 1, t1.getCacheTopology().getCurrentCH().getSegmentsForOwner(member).size());
```

Comment:
Since you know the owner indexes, you can assert them directly (maybe even extract this to a method):

      assertEquals(emptyList(), t1.getCacheTopology().getCurrentCH().locateOwnersForSegment(0));
      assertEquals(singletonList(address(1)), t1.getCacheTopology().getCurrentCH().locateOwnersForSegment(1));
      assertEquals(singletonList(address(2)), t1.getCacheTopology().getCurrentCH().locateOwnersForSegment(2));